src/ui/Analytics/AnalyticsClient.ts

```diff
@@ -115,8 +116,9 @@ export interface IAnalyticsClient {
    * ( `{}` ).
    * @param element The HTMLElement that the user has interacted with for this custom event. Default value is the
    * element on which the `Analytics` component is bound.
+   * @returns {Promise<Any>|Promise<IAPIAnalyticsEventResponse>|Promise<IAPIAnalyticsEventResponse[]>}
    */
-  logCustomEvent<TMeta>(actionCause: IAnalyticsActionCause, meta: TMeta, element: HTMLElement): void;
+  logCustomEvent<TMeta>(actionCause: IAnalyticsActionCause, meta: TMeta, element: HTMLElement): Promise<any>;
```

Comment:
Does it work if you return `Promise<Promise<IAPIAnalyticsEventResponse>|Promise<IAPIAnalyticsEventResponse[]>` instead ?